riot/runtime/scheduler/scheduler.ml

```diff
@@ -471,6 +486,34 @@ module Scheduler = struct
     Log.trace (fun f -> f "< exit worker loop")
 end
 
+module Blocking_scheduler = struct
+  (* include Scheduler *)
+  type t = blocking
+
+  let make sch domain = { scheduler = sch; domain }
+
+  let rec add_to_pool pool blocking =
+    let dom_list = Atomic.get pool.blocking_schedulers in
+    if
+      Atomic.compare_and_set pool.blocking_schedulers dom_list
+        (blocking :: dom_list)
+    then ()
+    else add_to_pool pool blocking
+
+  let rec remove_from_pool pool blocking =
+    let cur = Atomic.get pool.blocking_schedulers in
+    let without_removee = List.filter (fun sch -> sch.domain != blocking.domain) cur in
+    if Atomic.compare_and_set pool.blocking_schedulers cur without_removee then
+      ()
+    else remove_from_pool pool blocking
+
+
+  (* Override the handle exit function *)
+  (* let handle_exit_blocking_proc pool sch proc reason = *)
+    (* Scheduler.handle_exit_proc pool sch.scheduler proc reason; *)
+    (* remove_from_pool pool sch *)
```

Comment:
I was thinking I could somehow have a custom handle_exit function but just having the domain doesnt mean I can shutdown and remove the scheduler *I think*, because I really need to close it from inside the `run` + `run_loop` scheduler loop.